### ISSUE TITLE
Lost or repeated beats with Non-daw

### ISSUE THREAD

**Issue by aleeusgr** [hydrogen-music/hydrogen#602]
Hydrogen version * :1.0.0-beta1
Operating system + version : AVlinux 2018.4.2.
Audio driver + version : jackdmp version 1.9.12 tmpdir /dev/shm protocol 8
Non version 1.2.0
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Using hydrogen with NON-daw i get lost\double sample triggering.
debug shows:
```
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
```

Reproduce:
Start NonSession> start Non-daw and hydrogen, play. If hydrogen is in song mode - mismatch occurs on loop trigger in Non, if in pattern mode - every time pattern loops.
Swing setting does not affect the issue
The same issue I had with 0.9.7, 
 
I'm not 100% sure if its hydrogen issue, but I had same with ardour5&hydrogen and if I remember it right didn't have with qtractor&hydrogen.

**Comment by thijz** [comment]
just to clarify : 
Do you have H2 in jack transport slave mode and non-daw setup as jack transport master ?

**Comment by aleeusgr** [comment]
I cant find any way to change non-daw behaviour, it seems to Master itself by default. Hydrogen goes to slave automatically, i think. So yes, hydrogen is slave, non is master

**Comment by theGreatWhiteShark** [comment]
I can reproduce this on current master (commit 538eaeb).

1. Start the Jack server
2. Start an instance of Non-timeline and of Hydrogen (no need for Non-Session)

When hitting play (inside Hydrogen) with Hydrogen being *not in Jack Master-mode*, I get the following warnings:

```
Jack: JackClient::TransportLocate pos = 0
Jack: RequestNewPos pos = 0
(I) JackAudioDriver::stop jack_transport_stop()
(W) JackAudioDriver::locate Calling jack_transport_locate(0)
(I) JackAudioDriver::play jack_transport_start()
(I) ::int H2Core::audioEngine_start(bool, unsigned int) [audioEngine_start]
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(I) JackAudioDriver::relocateBBT ...
....
```

If Hydrogen is *in Jack Master-mode*, I get

```
Jack: JackClient::TransportLocate pos = 0
Jack: RequestNewPos pos = 0
(I) JackAudioDriver::stop jack_transport_stop()
(W) JackAudioDriver::locate Calling jack_transport_locate(0)
(I) ::void H2Core::audioEngine_stop(bool) [audioEngine_stop]
(W) JackAudioDriver::updateTransportInfo Frame offset mismatch; triggering resync in 2 cycles
(W) JackAudioDriver::relocateBBT Relocate: Call it off
```
and no further warning messages.

Not quite sure if this is an actual problem or if I'm able to grasp the Jack-driver, but I will look into it.

**Comment by theGreatWhiteShark** [comment]
The actual fix is contained in #678. I decided to kept it separate from the documentation.

**Comment by jeremyz** [comment]
Yes, that's my next reading ...

**Comment by theGreatWhiteShark** [comment]
Fixed in #950
